apps/web/modules/ee/quotas/actions.ts

```diff
@@ -0,0 +1,159 @@
+import { createQuota, deleteQuota, updateQuota } from "@/modules/ee/quotas/lib/quotas";
+import { z } from "zod";
+import { ZId } from "@formbricks/types/common";
+import { OperationNotAllowedError } from "@formbricks/types/errors";
+import { ZSurveyQuotaCreateInput, ZSurveyQuotaUpdateInput } from "@formbricks/types/quota";
+
+const ZDeleteQuotaAction = z.object({
+  quotaId: ZId,
+});
+
+export const deleteQuotaAction = authenticatedActionClient.schema(ZDeleteQuotaAction).action(
+  withAuditLogging(
+    "deleted",
+    "quota",
+    async ({
+      ctx,
+      parsedInput,
+    }: {
+      ctx: AuthenticatedActionClientCtx;
+      parsedInput: z.infer<typeof ZDeleteQuotaAction>;
+    }) => {
+      const isQuotasEnabled = await getIsQuotasEnabled();
+      if (!isQuotasEnabled) {
+        throw new OperationNotAllowedError("Quotas are not enabled");
+      }
+
+      const organizationId = await getOrganizationIdFromQuotaId(parsedInput.quotaId);
+      await checkAuthorizationUpdated({
+        userId: ctx.user.id,
+        organizationId,
+        access: [
+          {
+            type: "organization",
+            roles: ["owner", "manager"],
+          },
+          {
+            type: "projectTeam",
+            projectId: await getProjectIdFromQuotaId(parsedInput.quotaId),
+            minPermission: "readWrite",
+          },
+        ],
+      });
+
+      ctx.auditLoggingCtx.organizationId = organizationId;
+      ctx.auditLoggingCtx.quotaId = parsedInput.quotaId;
+
+      const result = await deleteQuota(parsedInput.quotaId);
+
+      ctx.auditLoggingCtx.oldObject = result;
+      return result;
+    }
+  )
+);
+
+const ZUpdateQuotaAction = z.object({
+  quotaId: ZId,
+  quota: ZSurveyQuotaUpdateInput,
+});
+
+export const updateQuotaAction = authenticatedActionClient.schema(ZUpdateQuotaAction).action(
+  withAuditLogging(
+    "updated",
+    "quota",
+    async ({
+      ctx,
+      parsedInput,
+    }: {
+      ctx: AuthenticatedActionClientCtx;
+      parsedInput: z.infer<typeof ZUpdateQuotaAction>;
+    }) => {
+      const isQuotasEnabled = await getIsQuotasEnabled();
+      if (!isQuotasEnabled) {
+        throw new OperationNotAllowedError("Quotas are not enabled");
+      }
+
+      const organizationId = await getOrganizationIdFromQuotaId(parsedInput.quotaId);
+      await checkAuthorizationUpdated({
+        userId: ctx.user.id,
+        organizationId,
+        access: [
+          {
+            type: "organization",
+            roles: ["owner", "manager"],
+          },
+          {
+            type: "projectTeam",
+            projectId: await getProjectIdFromQuotaId(parsedInput.quotaId),
+            minPermission: "readWrite",
+          },
+        ],
+      });
+
+      ctx.auditLoggingCtx.organizationId = organizationId;
+      const result = await updateQuota(parsedInput.quota, parsedInput.quotaId);
+      ctx.auditLoggingCtx.quotaId = parsedInput.quotaId;
+      ctx.auditLoggingCtx.newObject = result;
```

Comment:
🔧 **Fix**:  Add the quota as the oldObject, then we can audit log only the updated fields.

```suggestion
      ctx.auditLoggingCtx.oldObject = parsedInput.quota;
      ctx.auditLoggingCtx.newObject = result;
```